t/plugin/grpc-transcode4.t

```diff
@@ -0,0 +1,154 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use t::APISIX 'no_plan';
+# ensure that the JSON module of Perl is installed in your test environment.
+# If it is not installed, sudo cpanm JSON.
+use JSON;
+
+no_long_string();
+no_shuffle();
+no_root_location();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if (!$block->request) {
+        $block->set_value("request", "GET /t");
+    }
+});
+
+run_tests;
+
+__DATA__
+
+=== TEST 1: set rule
+--- config
+    location /t {
+       content_by_lua_block {
+          local http = require "resty.http"
+          local t = require("lib.test_admin").test
+          local code, body = t('/apisix/admin/protos/1',
+                ngx.HTTP_PUT,
+                [[{
+                   "content" : "syntax = \"proto3\";
+                    package user;
+                    service UserService {
+                        rpc GetUserInfo(UserRequest) returns (UserResponse) {}
+                     }
+
+                    enum Gender {
+                        GENDER_UNSPECIFIED = 0;
+                        GENDER_MALE = 1;
+                        GENDER_FEMALE = 2;
+                    }
+                    message Job {
+                        string items = 1;
```

Comment:
```suggestion
                        string name = 1;
```
this naming make me confused with `UserResponse.items` in test cases.

Reply:
I understand, you want to verify that the `fetch_proto_array_names` function can handle scenarios with fields of the same name at different levels.